src/providers/manoeuvres-by-category/manoeuvres-by-category.ts

```diff
@@ -0,0 +1,49 @@
+import { getManoeuvres as getManoeuvresB } from '../../modules/tests/test-data/cat-b/test-data.cat-b.selector';
+import {
+  CatCManoeuvres, CatCTestData,
+  getManoeuvres as getManoeuvresC,
+} from '../../modules/tests/test-data/cat-c/test-data.cat-c.selector';
+import {
+  CatDManoeuvres, CatDTestData,
+  getManoeuvres as getManoeuvresD,
+} from '../../modules/tests/test-data/cat-d/test-data.cat-d.selector';
+import { getManoeuvres as getManoeuvresBE } from '../../modules/tests/test-data/cat-be/test-data.cat-be.selector';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+import { Injectable } from '@angular/core';
+import { CategoryCode } from '@dvsa/mes-test-schema/categories/common';
+import { CatBEUniqueTypes } from '@dvsa/mes-test-schema/categories/BE';
+import { CatBUniqueTypes } from '@dvsa/mes-test-schema/categories/B';
+
+export type TestDataUnion =
+  | CatBUniqueTypes.TestData
+  | CatBEUniqueTypes.TestData
+  | CatCTestData
+  | CatDTestData;
+export type ManoeuvreUnion =
+  | CatBUniqueTypes.Manoeuvres
+  | CatBEUniqueTypes.Manoeuvres
+  | CatCManoeuvres
+  | CatDManoeuvres;
+
+@Injectable()
+export class ManoeuvresByCategoryProvider {
+
+  static getManoeuvresByCategoryCodeErrMsg: string = 'Error getting test category manoeuvres';
+
+  public getManoeuvresByCategoryCode(category: CategoryCode): (testData: TestDataUnion) => ManoeuvreUnion {
+    switch (category) {
+      case TestCategory.B: return getManoeuvresB;
+      case TestCategory.B1: return getManoeuvresB;
+      case TestCategory.BE: return getManoeuvresBE;
+      case TestCategory.C: return getManoeuvresC;
+      case TestCategory.C1: return getManoeuvresC;
+      case TestCategory.CE: return getManoeuvresC;
+      case TestCategory.C1E: return getManoeuvresC;
+      case TestCategory.D: return getManoeuvresD;
+      case TestCategory.D1: return getManoeuvresD;
+      case TestCategory.DE: return getManoeuvresD;
+      case TestCategory.D1E: return getManoeuvresD;
```

Comment:
this would be better and the same for cat C

`     
case TestCategory.D:
case TestCategory.D1:
case TestCategory.DE: 
case TestCategory.D1E:
     return getManoeuvresD;`

Reply:
let me know if that needs more explanation